shared/macros-oval.jinja

```diff
@@ -0,0 +1,199 @@
+{{%- if default_pass %}}
+{{%- set check_existence = "none_exist" -%}}
+{{%- set prefix_text = "value" -%}}
+{{%- set suffix_id = "_default_pass" -%}}
+{{%- else %}}
+{{%- set check_existence = "all_exist" -%}}
+{{%- set prefix_text = "absence" -%}}
+{{%- set suffix_id = "" -%}}
+{{%- endif %}}
+  <ind:textfilecontent54_test check="all" check_existence="{{{ check_existence }}}"
+  comment="tests the {{{ prefix_text }}} of {{{ parameter }}} setting in the {{{ path }}} file"
+  id="test_{{{ rule_id }}}{{{ suffix_id }}}" version="1">
+  <ind:object object_ref="obj_{{{ rule_id }}}{{{ suffix_id }}}" />
+  {{%- if not default_pass %}}
+  <ind:state state_ref="state_{{{ rule_id }}}{{{ suffix_id }}}" />
+  {{%- endif %}}
+  </ind:textfilecontent54_test>
+{{%- endmacro %}}
+
+{{#
+  Macro to check if a parameter in a configuration file is set (Object definition).
+  This function can take five parameters:
+    - path (String): Path to the configuration file to be checked.
+    - prefix_regex (String): Regular expression to be used in the beginning of the OVAL text file content check.
+    - parameter (String): The parameter to be checked in the configuration file.
+    - separator_regex (String): Regular expression to be used as the separator of parameter and value in a configuration file. If spaces are allowed, this should be included in the regular expression.
+    - default_pass (boolean): Implies that the value for the parameter is the default value. If set, the check will pass even if the parameter is not set in the configuration file.
+#}}
+{{%- macro oval_line_in_file_object(path='', prefix_regex='^[\s]*', parameter='', separator_regex='[\s]+', default_pass='') -%}}
+{{%- if default_pass %}}
+{{%- set suffix_id = "_default_pass" -%}}
+{{%- set regex = prefix_regex+parameter+separator_regex -%}}
+{{%- else %}}
+{{%- set suffix_id = "" -%}}
+{{%- set regex = prefix_regex+parameter+separator_regex+"(.*)[\s]*(?:|(?:#.*))?$" -%}}
+{{%- endif %}}
+  <ind:textfilecontent54_object id="obj_{{{ rule_id }}}{{{ suffix_id }}}" version="1">
+    <ind:filepath>{{{ path }}}</ind:filepath>
+    <ind:pattern operation="pattern match">{{{ regex }}}</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+{{%- endmacro %}}
+
+{{#
+  Macro to check if a expected value can be found in the extracted information of an OVAL object (State definition).
+  This function can take two parameters:
+    - value (String): The value to be checked. This can also be a regular expression (e.g: value1|value2 can match both values).
+    - multi_value (boolean): If set, it means that the parameter can accept multiple values and the expected value must be present in the current list of values.
+#}}
+{{%- macro oval_line_in_file_state(value='', multi_value='') -%}}
+{{%- if multi_value %}}
+{{%- set regex = "^.*"+value+".*$" -%}}
+{{%- else %}}
+{{%- set regex = "^"+value+"[\s]*$" -%}}
```

Comment:
Sorry for nitpicking this regex again.
Is `[\s]*` needed after the value?
As far as I understand, in line 115:
``` {{%- set regex = prefix_regex+parameter+separator_regex+"(.*)[\s]*(?:|(?:#.*))?$" -%}} ```
there is `(.*)[\s]*`, thus in the subexpression there shouldn't be any trailing white spaces.

Reply:
It makes sense. I'll remove it. Please don't be shy on pointing out nitpicks in those regular expressions, I am no expert in regex and they will probably need more tweaks